Samples/Notifications/Push/cpp-console-unpackaged/cpp-console-unpackaged.cpp

```diff
@@ -0,0 +1,203 @@
+            }
+            else if (args.status == PushNotificationChannelStatus::InProgressRetry)
+            {
+                LOG_HR_MSG(
+                    args.extendedError,
+                    "The WNS ChannelURI request is in back-off retry mode because of a retryable error! Expect delays in acquiring it. RetryCount = %d",
+                    args.retryCount);
+            }
+        });
+
+    auto result = co_await channelOperation;
+
+    if (result.Status() == PushNotificationChannelStatus::CompletedSuccess)
+    {
+        auto channel = result.Channel();
+
+        DateTimeFormatter formater = DateTimeFormatter(L"on {month.abbreviated} {day.integer(1)}, {year.full} at {hour.integer(1)}:{minute.integer(2)}:{second.integer(2)}");
+
+        std::cout << "\nWNS ChannelURI: " << winrt::to_string(channel.Uri().ToString()) << std::endl;
+        std::wcout << L"\nThe WNS ChannelURI will expire " << formater.Format(channel.ExpirationTime()).c_str() << std::endl;
+
+        // It's the caller's responsibility to keep the channel alive
+        co_return channel;
+    }
+    else if (result.Status() == PushNotificationChannelStatus::CompletedFailure)
+    {
+        LOG_HR_MSG(result.ExtendedError(), "We hit a critical non-retryable error with the WNS ChannelURI request!");
+        co_return nullptr;
+    }
+    else
+    {
+        LOG_HR_MSG(result.ExtendedError(), "Some other failure occurred.");
+        co_return nullptr;
+    }
+
+};
+
+winrt::Microsoft::Windows::PushNotifications::PushNotificationChannel RequestChannel()
+{
+    auto task = RequestChannelAsync();
+    if (task.wait_for(std::chrono::seconds(300)) != AsyncStatus::Completed)
+    {
+        task.Cancel();
+        return nullptr;
+    }
+
+    auto result = task.GetResults();
+    return result;
+}
+
+// Subscribe to an event which will get signaled whenever a foreground notification arrives.
+void SubscribeForegroundEventHandler(const winrt::Microsoft::Windows::PushNotifications::PushNotificationChannel& channel)
+{
+    winrt::event_token token = PushNotificationManager::Default().PushReceived([](auto const&, PushNotificationReceivedEventArgs const& args)
+        {
+            auto payload = args.Payload();
+
+            std::string payloadString(payload.begin(), payload.end());
+            std::cout << "\nPush notification content received from FOREGROUND: " << payloadString << std::endl;
+        });
+}
+
+int main()
+{
+    // An unpackaged app must initialize Dynamic Dependencies so we can consume the Windows App SDK APIs.
+    // See https://docs.microsoft.com/en-us/windows/apps/windows-app-sdk/tutorial-unpackaged-deployment?tabs=cpp,
+    // for details about using the WindowsAppSDK from an unpackaged app.
+    if (FAILED(LoadWindowsAppSDK()))
+    {
+        std::wcout << "\nCould not load Windows App SDK!" << std::endl;
+        std::cin.ignore();
+
+        return 1;
+    }
+
+    auto args = AppInstance::GetCurrent().GetActivatedEventArgs();
```

Comment:
Need to call Push/AppNotificationManager::Register before this so we can receive activation args from BI/Shell. Otherwise, the activation args won't be correctly routed to the GetActivatedEventArgs. In fact, if the app was activated and never called register, it will throw.

Reply:
Yup, lot has changed since this code was originally written. There was a time when you couldn't register a handler without first creating a channel...

fixed.